src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,130 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.EmptyDescException;
+import duke.exception.UnknownCommandException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+/**
+ * Parses the user's commands into a readable
+ * format for the other classes.
+ */
+public class Parser {
+
+    /**
+     * Represents all the valid commands that the user can input.
+     */
+    enum Commands {
+        FIND,
+        TODO,
+        DEADLINE,
+        EVENT,
+        MARK,
+        UNMARK,
+        DELETE,
+        LIST,
+        BYE
+
+    }
+
+
+
+    /**
+     * Matches the appropriate enum to the input converted
+     * into a string, then creates the relevant command to be
+     * executed. When an input is gibberish, an exception is
+     * also thrown.
+     *
+     * @param fullCommand String representation of user input.
+     * @return A command to be executed.
+     * @throws UnknownCommandException Error when input does not match
+     * any of the commands allowed.
+     * @throws EmptyDescException Input does not contain a description.
+     */
+    public static Command parse(String fullCommand) throws UnknownCommandException, EmptyDescException {
+        try {
+            String[] inputArr = fullCommand.split(" ");

```

Comment:
Perhaps you can consider doing split(" ", 2) to split the input into the key command word and its arguments? If you wish to go with this idea, please consider generating substrings of the argument string without the need to use Stringbuilder.

E.g. for deadline, we can generating description of task from argument by doing substring(0, "/by").